lnwire/custom.go

```diff
@@ -2,13 +2,62 @@ package lnwire
 
 import (
 	"bytes"
-	"errors"
+	"fmt"
 	"io"
 )
 
-// CustomTypeStart is the start of the custom type range for peer messages as
-// defined in BOLT 01.
-var CustomTypeStart MessageType = 32768
+var (
+	// CustomTypeStart is the start of the custom type range for peer
+	// messages as defined in BOLT 01.
+	CustomTypeStart MessageType = 32768
+
+	// CustomTypeOverride contains a set of message types < CustomTypeStart
+	// that lnd allows to be treated as custom messages. This allows us to
+	// override messages reserved for the protocol level and treat them as
+	// custom messages. This set of message types is stored as a global so
+	// that we do not need to pass around state when accounting for this
+	// set of messages in message creation.
+	//
+	// Note: This value should be written only once at startup, then used
+	// as a read-only set of message types.
```

Comment:
Went with a simple RW mutex + allowing the ability to call `SetCustomOverrides` rather than erroring out on multiple calls. Do you think we need to be that strict?

Reply:
will there ever be a case where we want to call `SetCustomOverrides` more than once throughout the lifetime of lnd? perhaps right? like if we wanted to add a new allowed custom message type via lncli and not need to restart lnd. In that case, this method will need to be changed so that it appends to the list of overrides & does not keep overwriting it. 
Non blocking though. Just a thought.

Reply:
sounds gooood